indico/modules/events/registration/client/js/form_setup/ItemSettingsModal.jsx

```diff
@@ -146,6 +146,38 @@ export default function ItemSettingsModal({id, sectionId, defaultNewItemType, on
             />
           )}
           {SettingsComponent && <SettingsComponent {...itemData} />}
+          {!meta.noRetentionPeriod && !fieldIsRequired && (
+            <>
+              <FinalInput
+                name="retentionPeriod"
+                type="number"
+                placeholder={Translate.string('Permanent')}
+                step="1"
+                min="1"
+                validate={v.optional(v.min(0))}
+                parse={val => (val === '' ? 0 : val)}
+                format={val => (val === 0 ? '' : val)}
```

Comment:
I'm not sure about this... we don't ever want `0` as the value here, it should be `null` when there's no retention period.

Also I noticed that changing this field and reverting to the previous value afterwards keeps the form "dirty" - you can easily notice this by changing the retention period to some new value and then back - the submit button becomes active, but then stays active even if you revert to the previous value.

I gave it a quick try and `parse={val => (val === '' ? null : +val)}` seems to fix the problem, but the server-side code shouldn't accept a `0` value at all...

PS: I noticed the same happens with `placesLimit`, and in the JSON we seem to send it as a string instead of a number... maybe that's why?

Reply:
With the `placesLimit` we agreed to use `0` as the "no limit" value, right? If so I can't find any problem with its behavior

Reply:
Yes I think so. I believe the problem is simply that we seem to keep strings there instead of converting to a number:

```patch
diff --git a/indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx b/indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx
index e8a22a74cd..7943aae1ef 100644
--- a/indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx
+++ b/indico/modules/events/registration/client/js/form/fields/BooleanInput.jsx
@@ -147,7 +147,7 @@ export function BooleanSettings() {
         step="1"
         min="1"
         validate={v.optional(v.min(0))}
-        parse={val => (val === '' ? 0 : val)}
+        parse={val => (val === '' ? 0 : +val)}
         format={val => (val === 0 ? '' : val)}
       />
       <FinalDropdown
diff --git a/indico/modules/events/registration/client/js/form/fields/CheckboxInput.jsx b/indico/modules/events/registration/client/js/form/fields/CheckboxInput.jsx
index db957da64b..0c5543a5aa 100644
--- a/indico/modules/events/registration/client/js/form/fields/CheckboxInput.jsx
+++ b/indico/modules/events/registration/client/js/form/fields/CheckboxInput.jsx
@@ -86,7 +86,7 @@ export function CheckboxSettings() {
       step="1"
       min="1"
       validate={v.optional(v.min(0))}
-      parse={val => (val === '' ? 0 : val)}
+      parse={val => (val === '' ? 0 : +val)}
       format={val => (val === 0 ? '' : val)}
     />
   );
```

With this patch applied I correctly get a number BTW. I think whoever wrote that original code (may have been me 🙈) thought number inputs already had a numeric `value`, but it's still a string of course...